lib/src/kbasesearchengine/main/IndexerWorker.java

```diff
@@ -169,18 +171,19 @@ public void stop(long millisToWait) throws InterruptedException {
         }
     }
     
-    private enum ErrorType {
+    // error disposition
+    private enum ErrorDisp {
         STD, FATAL, UNEXPECTED;
     }
     
-    private void logError(final ErrorType errtype, final Throwable e) {
+    private void logError(final ErrorDisp errtype, final Throwable e) {
         Utils.nonNull(errtype, "errtype");
         final String msg;
-        if (ErrorType.FATAL.equals(errtype)) {
+        if (ErrorDisp.FATAL.equals(errtype)) {
             msg = "Fatal error in indexer, shutting down";
-        } else if (ErrorType.STD.equals(errtype)) {
+        } else if (ErrorDisp.STD.equals(errtype)) {
             msg = "Error in indexer";
-        } else if (ErrorType.UNEXPECTED.equals(errtype)) {
+        } else if (ErrorDisp.UNEXPECTED.equals(errtype)) {
```

Comment:
Why not capture error code and message in the enum itself rather than having the two in separate places. 

```
public enum Error {
  FATAL(0, "Fatal error in indexer."),
  STD(1, "Standard error in indexer.");

  private final int code;
  private final String description;

  private Error(int code, String description) {
    this.code = code;
    this.description = description;
  }

  public String getDescription() {
     return description;
  }

  public int getCode() {
     return code;
  }

  @Override
  public String toString() {
    return code + ": " + description;
  }
}
```

Reply:
That's probably better, but this is out of scope for this PR. This enum is just a ternary selector for a general log error prefix string - if there were only two possibilities I would've just used a boolean. The only reason it shows up in this PR is because I renamed the enum to prevent a name conflict with the error type enum.

I'm not sure what the point of the code is - these aren't error types per se, they're just a ternary.

Reply:
I guess it's OK. But there is another private enum ErrorType in IndexerCoordinator that you may want to rename as ErrorDisp for consistency. To me they feel like error types, have the same values, and since they are being used in more than one place, I thought from a code quality perspective they could use some consolidation into an n-ary set of types with code and message as you have in ErrorType.java (only the message is in the form of a comment there).

Reply:
> But there is another private enum ErrorType in IndexerCoordinator that you may want to rename as ErrorDisp for consistency.

I forgot about that one. That one only had 2 states so I just got rid of it and used a boolean.

When I was writing the code, the enums were a) internal only and b) so trivially simple that I didn't worry about DRYing them out. 

I've refactored the old ErrorType class to better reflect its functionality.